src/main/java/org/commcare/formplayer/util/FormplayerDatadog.java

```diff
@@ -25,55 +34,116 @@ public String formatted() {
+        if (shouldAddTag(name)) {
+            Tag tag = new Tag(name, value);
+            requestScopedTags.put(name, tag);
+        }
+    }
+
+    /**
+     * Removes all request level tags
+     */
+    public void clearRequestScopedTags() {
+         requestScopedTags.clear();
     }
 
     // Datadog API Wrappers
-    public void recordExecutionTime(String aspect, final long value, List<Tag> additionalTags) {
-        List<String> tagsToSend = formattedTags(additionalTags);
+
+    /**
+     * Wrapper for StatsDClient's recordExecutionTime method
+     * @param aspect - name of the metric
+     * @param value - value of the metric
+     * @param transientTags - "one time" tags in addition to requestScopedTags
+     * NOTE: these tags have higher priority than request scoped tags (in the case of duplicate tag names)
+     */
+    public void recordExecutionTime(String aspect, final long value, List<Tag> transientTags) {
+        List<String> tagsToSend = getTagsToSend(transientTags);
         String[] datadogTags = tagsToSend.toArray(new String[tagsToSend.size()]);
         datadogClient.recordExecutionTime(aspect, value, datadogTags);
     }
 
-    public void increment(String aspect, List<Tag> additionalTags) {
-        List<String> tagsToSend = formattedTags(additionalTags);
+    /**
+     * Wrapper for StatsDClient's increment method
+     * @param aspect - name of the metric
+     * @param transientTags - "one time" tags in addition to requestScopedTags
+     * NOTE: these tags have higher priority than request scoped tags (in the case of duplicate tag names)
+     */
+    public void increment(String aspect, List<Tag> transientTags) {
+        List<String> tagsToSend = getTagsToSend(transientTags);
         String[] datadogTags = tagsToSend.toArray(new String[tagsToSend.size()]);
         datadogClient.increment(aspect, datadogTags);
     }
 
     // Private Helpers
-    private List<String> formattedTags(List<Tag> transientTags) {
+
+    /**
+     * Filters out tags that are not eligible for the current domain and formats
+     * @param transientTags - "one time" tags in addition to requestScopedTags
+     * NOTE: these tags have higher priority than request scoped tags (in the case of duplicate tag names)
+     * @return - list of formatted tags to send
+     */
+    private List<String> getTagsToSend(List<Tag> transientTags) {
         List<String> formattedTags = new ArrayList<>();
         // transient keys take precedence over existing keys
         HashSet<String> transientKeys = new HashSet<String>();
         
         for (Tag tag : transientTags) {
-            formattedTags.add(tag.formatted());
-            // keep track of transient tag names
-            transientKeys.add(tag.name);
+            if (shouldAddTag(tag.name)) {
+                formattedTags.add(tag.formatted());
+                // keep track of transient tag names
+                transientKeys.add(tag.name);
```

Comment:
I'm not sure which is better: not setting the tag or setting the tag to a predetermined null value. Just wanted to make sure you considered that, and here you chose that not setting the tag is better. This might show up in datadog as e.g. `form_name:N/A` or something like that, which is fine. Since it's not that there is no form name, you might choose to emit it as "form_name:_other" or something like that. But maybe that doesn't matter and this is simpler

Reply:
I think setting `form_name` to some specified value in this case is wise in order to differentiate intentionally not setting the form name vs unable to set the form name. I will make that change.